plutus-pab/src/Cardano/Wallet/RemoteClient.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes        #-}
+{-# LANGUAGE TypeApplications  #-}
+{-# LANGUAGE TypeOperators     #-}
+
+module Cardano.Wallet.RemoteClient
+    ( handleWalletClient
+    ) where
+
+import Cardano.Api.NetworkId.Extra (NetworkIdWrapper (..))
+import Cardano.Api.Shelley qualified as Cardano.Api
+import Cardano.Node.Types (MockServerConfig (..))
+import Control.Concurrent.STM qualified as STM
+import Control.Monad.Freer (Eff, LastMember, Member, type (~>))
+import Control.Monad.Freer.Error (Error, throwError)
+import Control.Monad.Freer.Reader (Reader, ask)
+import Control.Monad.IO.Class (MonadIO (..))
+import Data.Text qualified as Text
+import Plutus.Contract.Wallet (export)
+import Plutus.PAB.Core.ContractInstance.STM (InstancesState)
+import Plutus.PAB.Core.ContractInstance.STM qualified as Instances
+import Wallet.Effects (WalletEffect (..))
+import Wallet.Emulator.Error (WalletAPIError (..))
+import Wallet.Types (ContractInstanceId)
+
+handleWalletClient
+    :: forall m effs.
+    ( LastMember m effs
+    , MonadIO m
+    , Member (Error WalletAPIError) effs
+    , Member (Reader Cardano.Api.ProtocolParameters) effs
+    , Member (Reader InstancesState) effs
+    )
+    => MockServerConfig
+    -> Maybe ContractInstanceId
+    -> WalletEffect
+    ~> Eff effs
+handleWalletClient config cidM event = do
+    let NetworkIdWrapper networkId = mscNetworkId config
+    protocolParams <- ask @Cardano.Api.ProtocolParameters
+    case event of
+        OwnPubKeyHash -> do
+            error "Cardano.Wallet.RemoteClient: OwnPubKeyHash not yet implemented"
+
+        WalletAddSignature _ -> do
+            error "Cardano.Wallet.RemoteClient: WalletAddSignature not yet implemented"
```

Comment:
Use `throwError` so that they can be correctly handled in the `Contract`. Add a constructor for the error message: `RemoteClientFunctionNotSupported`